YAML/YAML.sublime-syntax

```diff
@@ -171,9 +171,17 @@ variables:
   _type_int_other: ([-+]?)([1-9][0-9_]*(?::[0-5]?[0-9])+) # (base 60)
 
   # http://yaml.org/type/float.html
-  _type_float_decimal: ([-+]?)((?:[0-9][0-9_]*)?(\.)[0-9.]*(?:[eE][-+][0-9]+)?)
+  _type_float_decimal: |-
+    (?x)
+    ([-+]?) (
+      (?:
+        (?:[0-9][0-9_]*) (\.) [0-9_]*
+      | (\.) (?:[0-9][0-9_]+)
```

Comment:
I'd suggest

```suggestion
      | (\.) [0-9_]+
```

to correctly highlight floating point numbers like `.1` with only one decimal. If we are concerned about `._` being highlighted as float per accident, we might  want to use 

```suggestion
      | (\.) (?:[0-9][0-9_]*)
```